src/core/components/inputs/TextInput.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {useEffect, useState} from 'react';
+import {StyledEyeBtn, StyledInputRowWrapper, StyledInputWrapper, StyledTextInput} from './Inputs.styled';
+import {
+    Keyboard,
+    KeyboardTypeOptions,
+    StyleProp,
+    TextStyle,
+    ViewStyle
+} from "react-native";
+import Ionicons from "@expo/vector-icons/Ionicons";
+import {RegularText} from "../textComponents/TextComponents";
+
+interface InputProps {
+    value: string,
+    required?: boolean,
+    keyboardType?: KeyboardTypeOptions,
+    autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters',
+    isEditable?: boolean,
+    multiline?: boolean,
+    maxLength?: number,
+    onChange: (arg0: string) => void,
+    errorHandler?: (arg0: boolean) => void,
+    label?: string,
+    placeholder?: string,
+    isPassword?: boolean,
+    validation?: { rule: RegExp, message: string }[],
+    styles?: {
+        wrapper?: StyleProp<ViewStyle>,
+        input?: StyleProp<TextStyle>,
+        label?: StyleProp<TextStyle>,
+        error?: StyleProp<TextStyle>
+    }
+}
+
```

Comment:
If you use standard props for native components, like multiline, maxLength, etc - you may all these props include in {... rest} param